packages/@uppy/core/src/Uppy.js

```diff
@@ -36,18 +36,21 @@ class Uppy {
   /** @type {Record<string, BasePlugin[]>} */
   #plugins = Object.create(null)
 
+  #uploader = new Uploader(
+    this.getState.bind(this),
+    this.setState.bind(this),
+    this.emit.bind(this),
+    this.log.bind(this),
+    this.getFile.bind(this),
+    () => this.opts,
```

Comment:
I feel strongly that we should pass the `Uppy` instance rather than try to cherry-pick what we use from it. What's the point of using classes if we are only passing methods around?
```suggestion
    this
```